app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,80 @@
+class RecipesController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @recipes = Recipe.all.where(user_id: current_user.id)
+  end
+
+  def show
+    @recipe = Recipe.includes(recipe_foods: [:food]).find_by(id: recipe_params[:recipe_id])
+  end
+
+  def toggle_visibility
+    @recipe = Recipe.find_by(id: toggle_visibility_params[:recipe_id])
+    @recipe.update_column(:public, toggle_visibility_params[:visibility] == 'true')
+    render json: { status: 'successful' }
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def ingredients
+    @recipe_id = recipe_params[:recipe_id]
+    @foods = Food.all
+  end
+
+  def add_ingredients
+    @recipe_food = RecipeFood.new(
+      quantity: add_ingredients_params[:quantity],
+      recipe_id: add_ingredients_params[:recipe_id],
+      food_id: add_ingredients_params[:food_id]
+    )
+
+    if @recipe_food.save
+      render json: { status: 'Food item successfully added' }, status: 201
+    else
+      render :ingredients, status: :unprocessable_entity
+    end
+  end
+
+  def remove_ingredient
+    recipe_food = RecipeFood.find_by(id: recipe_params[:recipe_food_id])
+    recipe_food.destroy
+    redirect_to recipe_path(recipe_params[:recipe_id])
```

Comment:
- [ ] @team its good practice to have both success and failure paths for methods that have  create, delete, or update  actions, kindly ensure your methods have both paths.

Reply:
Fixed.